vnext/Microsoft.ReactNative/IReactContext.idl

```diff
@@ -11,70 +11,158 @@ import "IReactPropertyBag.idl";
     String DebugBundlePath { get; };
+
+    DOC_STRING(
+      "A read-only snapshot of the @ReactInstanceSettings.BundleRootPath property value "
+      "at the time when the React instance was created.\n"
+      "Base path used for the location of the bundle.")
     String BundleRootPath { get; };
+
+    DOC_STRING(
+      "A read-only snapshot of the @ReactInstanceSettings.SourceBundleHost property value "
+      "at the time when the React instance was created.\n"
+      "When using @ReactInstanceSettings.UseFastRefresh, @ReactInstanceSettings.UseLiveReload, or @ReactInstanceSettings.UseWebDebugger this is the server hostname "
+      "that will be used to load the bundle from.")
     String SourceBundleHost { get; };
+
+    DOC_STRING(
+      "A read-only snapshot of the @ReactInstanceSettings.SourceBundlePort property value "
+      "at the time when the React instance was created.\n"
+      "When using @ReactInstanceSettings.UseFastRefresh, @ReactInstanceSettings.UseLiveReload, or @ReactInstanceSettings.UseWebDebugger this is the server port "
+      "that will be used to load the bundle from.")
     UInt16 SourceBundlePort { get; };
+
+    DOC_STRING(
+      "A read-only snapshot of the @ReactInstanceSettings.JavaScriptBundleFile property value "
+      "at the time when the React instance was created.\n"
+      "The name of the JavaScript bundle file to load. This should be a relative path from @.BundleRootPath. "
+      "The `.bundle` extension will be appended to the end, when looking for the bundle file.")
     String JavaScriptBundleFile { get; };
   }
 #endif
 
   [webhosthidden]
   DOC_STRING(
-    "The `IReactContext` object is given to native modules to communicate with other native modules, views, application, and the React Native instance. \n"
-    "It has the same lifetime as the React instance. When the React instance is reloaded or unloaded, the `IReactContext` is destroyed. \n"
-    "- Use the Properties to share native module's data with other components. \n"
-    "- Use the Notifications to exchange events with other components. \n"
-    "- Use @.CallJSFunction to call JavaScript functions, and @.EmitJSEvent to raise JavaScript events. \n"
-    "- Use @.UIDispatcher to schedule work in the UI thread. \n"
-    "- Use @.JSDispatcher to schedule work in the JavaScript thread.")
-  interface IReactContext {
+    "The `IReactContext` object is a weak pointer to the React instance. "
+    "It allows native modules and view managers to communicate with the application, and with other "
+    "native modules and view managers.\n"
+    "Since the @IReactContext is a weak pointer to the React instance, some of its functionality becomes unavailable "
+    "after the React instance is unloaded. When a React instance is reloaded inside of the @ReactNativeHost, the "
+    "previous React instance is unloaded and then a new React instance is created with a new @IReactContext.\n"
```

Comment:
> Not sure if possible, but it would be nice to show what it would look like or how to check if the weak pointer is unreferenced. And/or how to get it back.

We cannot dereference it because we never expose the wrapped up React instance. Though, the behavior of the IReactContext is going to change if we cannot dereference the React instance anymore. We need to document what will work or not work in such case.

Reply:
Added Issue #7506